rsmt2d_test.go

```diff
@@ -68,7 +67,84 @@ func TestEdsRepairRoundtripSimple(t *testing.T) {
 				// See extendeddatacrossword_test.go
 				t.Errorf("RepairExtendedDataSquare failed")
 			}
-			_ = repaired
+		})
+	}
+}
+
+func TestEdsRepairTwice(t *testing.T) {
+	bufferSize := 64
+	tests := []struct {
+		name string
+		// Size of each share, in bytes
+		shareSize int
+		codec     rsmt2d.Codec
+	}{
+		{"leopardFF8", bufferSize, rsmt2d.NewLeoRSFF8Codec()},
+		{"leopardFF16", bufferSize, rsmt2d.NewLeoRSFF16Codec()},
+		{"infectiousGF8", bufferSize, rsmt2d.NewRSGF8Codec()},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ones := bytes.Repeat([]byte{1}, bufferSize)
+			twos := bytes.Repeat([]byte{2}, bufferSize)
+			threes := bytes.Repeat([]byte{3}, bufferSize)
+			fours := bytes.Repeat([]byte{4}, bufferSize)
+
+			// Compute parity shares
+			eds, err := rsmt2d.ComputeExtendedDataSquare(
+				[][]byte{
+					ones, twos,
+					threes, fours,
+				},
+				tt.codec,
+				rsmt2d.NewDefaultTree,
+			)
+			if err != nil {
+				t.Errorf("ComputeExtendedDataSquare failed")
+			}
+
+			// Save all shares in flattened form.
+			flattened := make([][]byte, 0, eds.Width()*eds.Width())
+			for i := uint(0); i < eds.Width(); i++ {
+				flattened = append(flattened, eds.Row(i)...)
+			}
+
+			// Delete some shares, just enough so that repairing is possible, then remove one more.
+			missing := make([]byte, bufferSize)
+			copy(missing, flattened[1])
+			flattened[0], flattened[1], flattened[2], flattened[3] = nil, nil, nil, nil
+			flattened[4], flattened[5], flattened[6], flattened[7] = nil, nil, nil, nil
+			flattened[8], flattened[9], flattened[10] = nil, nil, nil
+			flattened[12], flattened[13] = nil, nil
+
+			// Repair square.
+			err = rsmt2d.RepairExtendedDataSquare(
+				eds.RowRoots(),
+				eds.ColRoots(),
+				flattened,
+				tt.codec,
+				rsmt2d.NewDefaultTree,
+			)
+			if err == nil {
+				// Should fail since insufficient data.
+				t.Errorf("RepairExtendedDataSquare did not fail")
```

Comment:
nit: It would be even nicer if the test would communicate which error you are expecting here (if feasible).

Reply:
Fixed in [6f587bc](https://github.com/celestiaorg/rsmt2d/pull/76/commits/6f587bc54aedbd39e7a2255707f2f6dcbd9e43a6)